### PR TITLE
Conduit-plugin [3]: Internal changes

### PR DESCRIPTION
**Pull request by lovromazgon**
### Description

This ties the new plugin registry into the existing Conduit code. This PR breaks existing plugins, they need to be adjusted to use the new SDK.

NB: Tests fail because tests use the file plugin that's not using the new SDK yet. This is fixed in #88.

Related to #65

### Quick checks:

- [X] I have followed the [Code Guidelines](https://github.com/ConduitIO/conduit/blob/main/docs/code_guidelines.md).
- [X] There is no other [pull request](https://github.com/ConduitIO/conduit/pulls) for the same update/change.
- [X] I have written unit tests.
- [X] I have made sure that the PR is of reasonable size and can be easily reviewed.

### CODE REVIEW

File: pkg/connector/source.go [hunk 1]
```diff
@@ -83,86 +79,89 @@ func (s *source) SetState(state SourceState) {
...
-	s.client = client
-	s.plugin = plug
+	s.plugin = src
 	s.persister.ConnectorStarted()
 	return nil
 }
 
+func (s *source) Stop(ctx context.Context) error {
+	if !s.IsRunning() {
+		return cerrors.New("plugin is not running")
```
Comment (hariso) [inline pkg/connector/source.go:138]: Maybe plugin.ErrPluginNotRunning?

**Review by hariso — APPROVED** [review]
LGTM! I have a question and a couple of nits, nothing blocking. 👍

File: pkg/connector/source.go [hunk 2]
```diff
@@ -171,20 +170,29 @@ func (s *source) Teardown(ctx context.Context) error {
...
 	}
-	s.lastPosition = r.Position
+
+	if r.Key == nil {
+		r.Key = record.RawData{}
+	}
+	if r.Payload == nil {
+		r.Payload = record.RawData{}
+	}
+	r.ReadAt = time.Now().UTC() // TODO now that records can be read asynchronously, should we move this to the plugin SDK?
```
[the remark this answers is not in the document]
Comment (lovromazgon) [reply pkg/connector/source.go:187]: Yet another issue to be opened, it's on my list 😅

File: pkg/connector/destination.go [hunk 3]
```diff
@@ -78,81 +78,78 @@ func (s *destination) SetState(state DestinationState) {
...
 	return nil
 }
 
 func (s *destination) Teardown(ctx context.Context) error {
 	if !s.IsRunning() {
-		return plugins.ErrNotRunning
+		return plugin.ErrPluginNotRunning
 	}
 
+	// TODO call Stop separately before Teardown. The idea is to let the plugin
```
[the remark this answers is not in the document]
Comment (hariso) [reply pkg/connector/destination.go:140]: Yup, let's then discuss this further in that issue. 👍

[the remark this answers is not in the document]
Comment (lovromazgon) [reply pkg/connector/destination.go:140]: Here it is: https://github.com/ConduitIO/conduit/issues/222

**Pull request merged**